prusti-viper/src/encoder/mir/procedures/encoder.rs

```diff
@@ -312,7 +474,42 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
                 self.encode_assign_operand(block_builder, location, encoded_target, operand)?;
             }
             // mir::Rvalue::Repeat(Operand<'tcx>, Const<'tcx>),
-            // mir::Rvalue::Ref(Region<'tcx>, BorrowKind, Place<'tcx>),
+            mir::Rvalue::Ref(region, borrow_kind, place) => {
+                // TODO: create proper ref, add "is_mut" and lifetime/region
+                let _is_mut = matches!(
+                    borrow_kind,
+                    mir::BorrowKind::Mut {
+                        allow_two_phase_borrow: _,
+                    }
+                );
+                let encoded_place = self.encoder.encode_place_high(self.mir, *place)?;
+                let rd_perm: u32 = self.read_permission_amount();
+                let region_name = region.lifetime_string();
+                let encoded_rvalue = vir_high::Rvalue::ref_(
+                    encoded_place,
+                    region_name.clone(),
+                    rd_perm,
+                    encoded_target.clone(),
+                );
+                let assign_statement = vir_high::Statement::assign(
+                    encoded_target.clone(),
+                    encoded_rvalue,
+                    self.register_error(location, ErrorCtxt::Assign),
+                );
+                block_builder.add_statement(self.set_statement_error(
+                    location,
+                    ErrorCtxt::Assign,
+                    assign_statement,
+                )?);
+                // TODO: check here if borrow or reborrow?
+                self.encode_rvalue_borrow(
```

Comment:
Isn't this subsumed by `encoded_rvalue` above?

Reply:
Currently, `add_rvalue_borrow` adds the `borrow(lftA, q, _Y.ref)` statement but
depending on `encoded_place` in `encoded_rvalue`, we will need a `reborrow(lftB,
lftC, q, _Y.ref)`.

Reply:
I see.